src/Microsoft.OpenApi.CSharpAnnotations.DocumentGeneration/DocumentConfigFilters/DocumentVariantAttributesFilter.cs

```diff
@@ -30,70 +31,86 @@ public class DocumentVariantAttributesFilter : IDocumentConfigFilter
         /// Thrown when there is a conflict
         /// between the attributes in the existing document variant info and the new one.
         /// </exception>
-        public void Apply(
+        /// <returns>The list of generation errors, if any produced when processing the filter."></returns>
+        public IList<GenerationError> Apply(
             IDictionary<DocumentVariantInfo, OpenApiDocument> documents,
             XElement element,
             IList<XDocument> xmlDocuments,
             DocumentConfigFilterSettings settings)
         {
-            var variantElements = element.Elements(KnownXmlStrings.Variant);
-            
-            foreach (var variantElement in variantElements)
+            var generationErrors = new List<GenerationError>();
+
+            try
             {
-                // First, populate the document variant info attributes using the document variant option.
-                var variantOptions = variantElement.Element(KnownXmlStrings.Options)?.Elements(KnownXmlStrings.Option);
+                var variantElements = element.Elements(KnownXmlStrings.Variant);
```

Comment:
do you need a null check on the element?

Reply:
sure will do on all parameters

---
In reply to: [284036413](https://github.com/microsoft/OpenAPI.NET.CSharpAnnotations/pull/208#discussion_r284036413) [](ancestors = 284036413)